io/src/main/scala/sbt/io/PathMapper.scala

```diff
@@ -112,12 +112,10 @@ abstract class Mapper {
    * Selects descendants of `base` directory matching `filter` and maps them to a path relative to `base`.
    * `base` itself is not included.
    */
-  def selectSubpaths(base: File, filter: FileFilter): Traversable[(File, String)] = {
-    val path = base.toPath
+  def selectSubpaths(base: File, filter: FileFilter): Traversable[(File, String)] =
     PathFinder(base).globRecursive(filter).get().collect {
-      case f if f != path => f -> path.relativize(f.toPath).toString
+      case f if f != base => f -> base.toPath.relativize(f.toPath).toString
```

Comment:
I wonder if there could have been some sort of linting that could've caught this.

Reply:
Yeah. I'm sort of surprised that I didn't get an unrelated type warning. From what I can tell, this seems baked into scala2: https://github.com/lampepfl/dotty/issues/1247. I wonder if there is a compiler plugin though that can catch this kind of thing.

Reply:
It does feel like help should be nearby.
```
val f = new java.io.File("f"); val p = f.toPath; f match { case `p` => }
```
By coincidence, I just tweaked a [similar expression](https://github.com/scala/scala/pull/8598/commits/4af22b3267a3646dfd182fd196626aedd20bc716#diff-5896f2db23c2c8679f9b2bb44c18adc1R175) albeit different file and path. I tend to stick to `nio.file.Path` now. There is a gotcha in that `p.endsWith("f")` is not `String::endsWith`.